internal/controller/clusterdeployment_reconciler.go

```diff
@@ -0,0 +1,225 @@
+		hivev1.ClusterInstallCompletedClusterDeploymentCondition,
+		hivev1.ClusterInstallFailedClusterDeploymentCondition,
+		hivev1.ClusterInstallStoppedClusterDeploymentCondition,
+	}
+}
+
+func updateSCProvisionedStatus(cd *hivev1.ClusterDeployment, sc *v1alpha1.SiteConfig) {
+	// Check if cluster has finished installing, if it has -> update siteConfig.Status.Conditions(Provisioned -> Completed)
+	if cd.Spec.Installed {
+		conditions.SetStatusCondition(&sc.Status.Conditions,
+			conditions.Provisioned,
+			conditions.Completed,
+			metav1.ConditionTrue,
+			"Provision completed")
+	} else if installStopped := conditions.FindConditionType(cd.Status.Conditions, hivev1.ClusterInstallStoppedClusterDeploymentCondition); installStopped != nil {
+		// Check if siteconfig.Status Provisioned -> InProgress condition
+		if found := meta.FindStatusCondition(sc.Status.Conditions, string(conditions.Provisioned)); found == nil {
+			if !cd.Spec.Installed && installStopped.Status == corev1.ConditionStatus(metav1.ConditionFalse) {
+				conditions.SetStatusCondition(&sc.Status.Conditions,
+					conditions.Provisioned,
+					conditions.InProgress,
+					metav1.ConditionTrue,
+					"Provisioning cluster")
+			}
+		}
+	}
+}
+
+func updateSCDeploymentConditions(cd *hivev1.ClusterDeployment, sc *v1alpha1.SiteConfig) {
+	// Compare siteConfig.Status.installConditions to clusterDeployment.Conditions
+	for _, cond := range clusterInstallConditionTypes() {
+		installCond := conditions.FindConditionType(cd.Status.Conditions, cond)
+		if installCond == nil {
+			// not found, initialize with Unknown fields
+			installCond = &hivev1.ClusterDeploymentCondition{
+				Type:    cond,
+				Status:  corev1.ConditionUnknown,
+				Reason:  "Unknown",
+				Message: "Unknown"}
+		}
+
+		now := metav1.NewTime(time.Now())
+
+		// Search SiteConfig status DeploymentConditions for the installCond
+		scCond := conditions.FindConditionType(sc.Status.DeploymentConditions, installCond.Type)
+		if scCond == nil {
+			installCond.LastTransitionTime = now
+			installCond.LastProbeTime = now
+			sc.Status.DeploymentConditions = append(sc.Status.DeploymentConditions, *installCond)
+		} else {
+			scCond.Status = installCond.Status
+			scCond.Reason = installCond.Reason
+			scCond.Message = installCond.Message
+			scCond.LastProbeTime = now
+
+			if scCond.Status != installCond.Status {
+				scCond.LastTransitionTime = now
+			}
+		}
+	}
+}
+
+func isOwnedBySiteConfig(ownerRefs []metav1.OwnerReference) bool {
+	for _, ownerRef := range ownerRefs {
+		if ownerRef.Kind == v1alpha1.SiteConfigKind {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *ClusterDeploymentReconciler) getSiteConfig(ctx context.Context, cd *hivev1.ClusterDeployment) (*v1alpha1.SiteConfig, error) {
+	siteConfigRef := ""
+	for _, ownerRef := range cd.GetOwnerReferences() {
+		if ownerRef.Kind == v1alpha1.SiteConfigKind {
+			siteConfigRef = ownerRef.Name
+			break
+		}
+	}
```

Comment:
This can be deduplicated:

```suggestion
func siteConfigOwner(ownerRefs []metav1.OwnerReference) string {
	for _, ownerRef := range ownerRefs {
		if ownerRef.Kind == v1alpha1.SiteConfigKind {
			return ownerRef.Name
		}
	}
	return ""
}

func isOwnedBySiteConfig(ownerRefs []metav1.OwnerReference) bool {
	return siteConfigOwner(ownerRefs) != ""
}

func (r *ClusterDeploymentReconciler) getSiteConfig(ctx context.Context, cd *hivev1.ClusterDeployment) (*v1alpha1.SiteConfig, error) {
	siteConfigRef := siteConfigOwnerRef(cd.GetOwnerReferences())
```